support/src/indirect_intrusive_base.h

```diff
@@ -0,0 +1,27 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+/*
+ * Copyright (C) 2016 Red Hat Inc.
+ */
+
+
+#pragma once
+
+#include <memory>
+#include <vector>
+#include <string>
+#include <iostream>
+#include <functional>
+
+#include "assert.h"
+
+
+namespace crimson {
+  using IndIntruHeapData = size_t;
+
+  template<typename I, typename T, IndIntruHeapData T::*heap_info,  typename ... Cs>
+  class IndIntruBase {
```

Comment:
This class will come handy if we decide to put virtual heap methods such as top(), empty(), adjust(), etc., plus iterator, const iterator.